app/src/main/java/com/simply/birthdayapp/main/MainScreen.kt

```diff
@@ -43,13 +45,23 @@ fun MainScreen(
             startDestination = BottomNavBarDestination.HomeDestination
         ) {
             composable<BottomNavBarDestination.HomeDestination> {
-                HomeMainScreen()
+                HomeMainScreen() { mode ->
+                    navController.navigate(BottomNavBarDestination.AddEventDestination(BirthdayMode.Edit(mode.birthday)))
+                }
             }
             composable<BottomNavBarDestination.ShopDestination> {
                 ShopMainScreen()
             }
-            composable<BottomNavBarDestination.AddEventDestination> {
-                AddEventScreen(navigateToMain = navigateToMain)
+            composable<BottomNavBarDestination.AddEventDestination>(
+                typeMap = mapOf(
+                    typeOf<Birthday>() to HomeNavType.HomeDomainType,
+                    typeOf<BirthdayMode>() to BirthdayModeNavType.BirthdayModeType
+                )
+            ) {
+                val eventMode = it.toRoute<BottomNavBarDestination.AddEventDestination>().eventMode
```

Comment:
consider to use more meaningful names for example argBirthdayMode instead of eventMode